.github/workflows/unit_tests.yml

```diff
@@ -40,7 +40,7 @@ jobs:
       - uses: actions/cache@v1
         with:
           path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('requirements/*.txt') }}
+          key: ${{ runner.os }}-pip-${{ hashFiles('pyproject.toml') }}
```

Comment:
Move these to setup-python

Reply:
Could you elaborate?

Reply:
`setup-python` action is built-in: https://github.com/actions/setup-python/#caching-packages-dependencies

Reply:
done, also upgraded the versions of a few actions